presto-spi/src/main/java/io/prestosql/spi/connector/ConnectorMetadata.java

```diff
@@ -806,4 +807,14 @@ default ConnectorTableProperties getTableProperties(ConnectorSession session, Co
     {
         return Optional.empty();
     }
+
+    /**
+     * Canonicalizes the provided SQL identifier according to connector-specific rules
+     * for the purpose of providing the name in metadata APIs
+     */
+    default String canonicalize(ConnectorSession session, String identifier, boolean delimited)
```

Comment:
I think we can skip passing session here. The way how remote data source is typically static and it should not change at runtime.
Also this will make this change easier. If we ever hit a use case where session is needed we can always add it then, it should be easy change.